pkg/steps/multi_stage.go

```diff
@@ -367,6 +367,10 @@ func (s *multiStageTestStep) runSteps(
 				}
 				if err := s.client.Delete(cleanupCtx, &pod); err != nil && !kerrors.IsNotFound(err) {
 					errs = append(errs, fmt.Errorf("failed to delete pod %s with label %s=%s: %w", pod.Name, MultiStageTestLabel, s.name, err))
+				} else {
+					if err := waitForPodDeletion(s.client, s.jobSpec.Namespace(), pod.Name, pod.UID); err != nil {
```

Comment:
Using `continue` above  would allow us to reduce indent level and IMO the block would "flow" better (there's a similar `if cond { continue }` statement above.

Reply:
Sounds good. Change pushed.